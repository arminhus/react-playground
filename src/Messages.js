import React from "react";
import "./Messages.css";

let Messages = props => {
  const badge = props.unread ? (
    <div className="unread_count badge">{props.unread}</div>
  ) : null;
  return (
    <div className="messages">
      {props.name}
      {badge}
    </div>
  );
};

export default Messages;
