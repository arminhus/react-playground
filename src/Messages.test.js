import React from "react";
import ReactDom from "react-dom";
import Messages from "./Messages";
import renderer from "react-test-renderer";

describe("Message component", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDom.render(<Messages />, div);
    ReactDom.unmountComponentAtNode(div);
  });

  it("renders the UI as expected", () => {
    const tree = renderer
      .create(<Messages name="Messages" unread={4} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("should render ui without crashing when unread", () => {
    const tree = renderer
      .create(<Messages name="Messages" unread={0} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
