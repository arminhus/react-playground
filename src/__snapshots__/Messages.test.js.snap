// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message component renders the UI as expected 1`] = `
<div
  className="messages"
>
  Messages
  <div
    className="unread_count badge"
  >
    4
  </div>
</div>
`;

exports[`Message component should render ui without crashing when unread 1`] = `
<div
  className="messages"
>
  Messages
</div>
`;

exports[`renders the UI as expected 1`] = `
<div
  className="messages"
>
  Messages
  <div
    className="unread_count badge"
  >
    4
  </div>
</div>
`;

exports[`should render ui without crashing when unread 1`] = `
<div
  className="messages"
>
  Messages
</div>
`;
